server/controllers/articles.js

```diff
@@ -214,6 +221,59 @@ class ArticleController {
       next(err);
     }
   }
+
+  /**
+   * Get article ratings
+   * @async
+   * @param  {object} req - Request object
+   * @param {object} res - Response object
+   * @param {object} next The next middleware
+   * @return {json} Returns json object
+   * @static
+   */
+  static async getArticleRatings(req, res, next) {
+    try {
+      const { slug } = req.params;
+      const { page, search } = req.query;
+      const paginate = new Pagination(page, req.query.limit);
+      const { limit, offset } = await paginate.getQueryMetadata();
+
+      // Get the articleId
+      const article = await Article.findOne({ where: { slug } });
+
+      if (!article) return Response.error(res, 404, 'Article does not exist');
+
+      const { id: articleId } = article.dataValues;
+      const ratingsDetails = await ratingsPayload(articleId);
```

Comment:
this is totally not necessary

you only need to count

`const count = Rating.count({where: {articleId}});`

and then get ratings
```javascript
const ratings = Rating.findAll({ 
  where: { articleId },
  limit,
  offset,
...
```

you should use same step for article

Reply:
Thanks for the feedback, was really helpful